File: EASSET.Web/Modules/MasterData/UKPB/UKPBForm.cs
using Serenity.ComponentModel;
using System;

namespace EASSET.MasterData.Forms;

[FormScript("MasterData.UKPB")]
[BasedOnRow(typeof(UKPBRow), CheckNames = true)]
public class UKPBForm
{
    public string UnitId { get; set; }
    public string SubUnitId { get; set; }
    public string Upbid { get; set; }
    public string UkpbKode { get; set; }
    public string UkpbNama { get; set; }
    //public string ThAng { get; set; }
    [TextAreaEditor(Rows = 5, Cols =10)]
    public string UkpbAlamat { get; set; }
  
    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public short Dlt { get; set; }
}

File: EASSET.Web/Modules/ServerTypes/MasterData/UKPBForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { ServiceLookupEditor, StringEditor, TextAreaEditor } from "serenity.corelib";

export interface UKPBForm {
    UnitId: ServiceLookupEditor;
    SubUnitId: ServiceLookupEditor;
    Upbid: ServiceLookupEditor;
    UkpbKode: StringEditor;
    UkpbNama: StringEditor;
    UkpbAlamat: TextAreaEditor;
}

export class UKPBForm extends PrefixedContext {
    static readonly formKey = 'MasterData.UKPB';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UKPBForm.init)  {
            UKPBForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;

            initFormType(UKPBForm, [
                'UnitId', w0,
                'SubUnitId', w0,
                'Upbid', w0,
                'UkpbKode', w1,
                'UkpbNama', w1,
                'UkpbAlamat', w2
            ]);
        }
    }
}